npe2/manifest/schema.py

```diff
@@ -147,14 +147,47 @@ def _check_engine_version(cls, values: dict) -> dict:
             )
         return values
 
+    @validator("display_name")
+    def validate_display_name(cls, v):
+
+        regex = r"^[^\W_][\w -~]{1,38}[^\W_]$"
+        if not bool(re.match(regex, v)):
+            raise ValueError(
+                f"{v} is not a valid display_name.  The display_name must "
+                "be 3-40 characters long, containing printable word characters, "
+                "and must not begin or end with an underscore, white space, or "
+                "non-word character."
+            )
+        return v
+
     @root_validator
     def _validate_root(cls, values: dict) -> dict:
         invalid_commands = []
+        invalid_python_names = []
         if values.get("contributions") is not None:
             for command in values["contributions"].commands or []:
-                if not command.id.startswith(values["name"]):
+                id_start_actual = command.id.split(".")[0]
+                if values["name"] != id_start_actual:
                     invalid_commands.append(command.id)
 
+                try:
+                    # check for normalized version of python name that matches
+                    # package name. Does it need to be lowercase?
+                    python_name_start_expected = "".join(
+                        [i.lower() if i != "-" else "_" for i in values["name"]]
+                    )
+                    python_name_start_actual = command.python_name.split(".")[0]
+                    python_name_start_actual = python_name_start_actual.split(":")[0]
+
+                    if (python_name_start_expected != "dynamic") and (
+                        python_name_start_expected != python_name_start_actual
+                    ):
+                        invalid_python_names.append(command.python_name)
+
+                except (NameError, AttributeError):
+                    # There isn't a python_name here.
+                    pass
+
```

Comment:
```suggestion
```

I don't think we want this.  As far as I know, the python_name might not have anything to do with the package/plugin name.

I could be wrong! so let me know what you think.